pkg/ipamclaimswebhook/podmutator.go

```diff
@@ -153,10 +189,17 @@ func (a *IPAMClaimsValet) Handle(ctx context.Context, request admission.Request)
 		podNetworkSelectionElements = append(podNetworkSelectionElements, *networkSelectionElement)
 	}
 
-	if len(podNetworkSelectionElements) > 0 {
-		newPod, err := podWithUpdatedSelectionElements(pod, podNetworkSelectionElements)
-		if err != nil {
-			return admission.Errored(http.StatusInternalServerError, err)
+	if len(podNetworkSelectionElements) > 0 || primaryNetworkIpamClaimName != "" {
+		newPod := pod.DeepCopy()
+		if len(podNetworkSelectionElements) > 0 {
+			var err error
+			newPod, err = podWithUpdatedSelectionElements(newPod, podNetworkSelectionElements)
+			if err != nil {
+				return admission.Errored(http.StatusInternalServerError, err)
+			}
+		}
+		if primaryNetworkIpamClaimName != "" {
+			newPod.Annotations[config.OVNPrimaryNetworkIPAMClaimAnnotation] = primaryNetworkIpamClaimName
 		}
```

Comment:
nit: please consider refactoring this
maybe by doing line 194 (due to `len(podNetworkSelectionElements) > 0`) and line 201 (due to `rimaryNetworkIpamClaimName != ""`)
separately, and marking on both that a change occurred, and then 
doing the common code of line 205 if change occurred ?

Reply:
Refactored a little so to reduce the networkselection elements copy and the condition check and make easier to follow the logic.